gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -311,6 +311,19 @@ public List<String> deriveAllRemoteNames() {
     return List.ofAll(jgitRepoForMainGitDir.getRemoteNames());
   }
 
+  @Override
+  public Option<String> deriveRebasedBranch() throws GitCoreException {
+    Option<Path> headNamePath = Stream.of("rebase-apply", "rebase-merge")
+        .map(dir -> jgitRepoForWorktreeGitDir.getDirectory().toPath().resolve(dir).resolve("head-name"))
+        .find(path -> path.toFile().isFile());
+    return !headNamePath.isEmpty()
```

Comment:
Nit: `find(<condition>)` + `!isEmpty()` can probably be replaced with sth like `exists(<condition>)`

Reply:
Whoops no, ignore the comment